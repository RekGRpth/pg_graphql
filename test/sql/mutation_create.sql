begin;

    create table account(
        id serial primary key,
        email varchar(255) not null
    );

    insert into public.account(email)
    values
        ('aardvark@x.com');

    create table blog(
        id serial primary key,
        owner_id integer not null references account(id)
    );

    -- Make sure functions still work
    create function _echo_email(account)
        returns text
        language sql
    as $$ select $1.email $$;


    select graphql.resolve($$
    mutation {
      createAccount(object: {
        email: "foo@barsley.com"
      }) {
        id
        echoEmail
        blogCollection {
            totalCount
        }
      }
    }
    $$);

    select * from account;


    select graphql.resolve($$
    mutation {
      createBlog(object: {
        ownerId: 2
      }) {
        id
        owner {
          id
        }
      }
    }
    $$);

    select * from blog;



rollback;
