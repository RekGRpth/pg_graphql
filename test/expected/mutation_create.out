begin;
    create table account(
        id serial primary key,
        email varchar(255) not null
    );
    insert into public.account(email)
    values
        ('aardvark@x.com');
    create table blog(
        id serial primary key,
        owner_id integer not null references account(id)
    );
    -- Make sure functions still work
    create function _echo_email(account)
        returns text
        language sql
    as $$ select $1.email $$;
    select graphql.resolve($$
    mutation {
      createAccount(object: {
        email: "foo@barsley.com"
      }) {
        id
        echoEmail
        blogCollection {
            totalCount
        }
      }
    }
    $$);
                                                          resolve                                                          
---------------------------------------------------------------------------------------------------------------------------
 {"data": {"createAccount": {"id": 2, "echoEmail": "foo@barsley.com", "blogCollection": {"totalCount": 0}}}, "errors": []}
(1 row)

    select * from account;
 id |      email      
----+-----------------
  1 | aardvark@x.com
  2 | foo@barsley.com
(2 rows)

    select graphql.resolve($$
    mutation {
      createBlog(object: {
        ownerId: 2
      }) {
        id
        owner {
          id
        }
      }
    }
    $$);
                                resolve                                
-----------------------------------------------------------------------
 {"data": {"createBlog": {"id": 1, "owner": {"id": 2}}}, "errors": []}
(1 row)

    select * from blog;
 id | owner_id 
----+----------
  1 |        2
(1 row)

rollback;
